test/SlowTests/Issues/RavenDB_14939.cs

```diff
@@ -0,0 +1,252 @@
+                }
+            }))
+            {
+                store.ExecuteIndex(new MyIndex_WithoutAnalyzer());
+
+                Fill(store);
+
+                var errors = WaitForIndexingErrors(store);
+                Assert.Equal(1, errors.Length);
+                Assert.Equal(1, errors[0].Errors.Length);
+                Assert.Contains("Cannot find analyzer type 'MyAnalyzer' for field: @default", errors[0].Errors[0].Error);
+
+                store.Maintenance.Send(new PutAnalyzersOperation(new AnalyzerDefinition
+                {
+                    Name = "MyAnalyzer",
+                    Code = GetAnalyzer("RavenDB_14939.MyAnalyzer.cs")
+                }));
+
+                store.Maintenance.Send(new ResetIndexOperation(new MyIndex_WithoutAnalyzer().IndexName));
+
+                AssertCount<MyIndex_WithoutAnalyzer>(store);
+            }
+        }
+
+        private static void Fill(IDocumentStore store)
+        {
+            using (var session = store.OpenSession())
+            {
+                session.Store(new Customer() { Name = "Rogério" });
+                session.Store(new Customer() { Name = "Rogerio" });
+                session.Store(new Customer() { Name = "Paulo Rogerio" });
+                session.Store(new Customer() { Name = "Paulo Rogério" });
+                session.SaveChanges();
+            }
+        }
+
+        private static void AssertCount<TIndex>(IDocumentStore store)
+            where TIndex : AbstractIndexCreationTask, new()
+        {
+            WaitForIndexing(store);
+
+            using (var session = store.OpenSession())
+            {
+                var results = session.Query<Customer, TIndex>()
+                    .Customize(x => x.NoCaching())
+                    .Search(x => x.Name, "Rogério*");
+
+                Assert.Equal(results.Count(), 4);
+            }
+        }
+
+        private static string GetAnalyzer(string name)
+        {
+            using (var stream = GetDump(name))
+            using (var reader = new StreamReader(stream))
+                return reader.ReadToEnd();
+        }
+
+        private static Stream GetDump(string name)
+        {
+            var assembly = typeof(RavenDB_14939).Assembly;
+            return assembly.GetManifestResourceStream("SlowTests.Data." + name);
+        }
+
+        private class Customer
+        {
+            public string Id { get; set; }
+            public string Name { get; set; }
+        }
+
+        private class MyIndex : AbstractIndexCreationTask<Customer>
+        {
+            public MyIndex()
+            {
+                Map = customers => from customer in customers
+                                   select new
+                                   {
+                                       Name = customer.Name
+                                   };
+
+                Indexes.Add(x => x.Name, FieldIndexing.Search);
+                Analyzers.Add(x => x.Name, "MyAnalyzer");
```

Comment:
How is the index name determined?

Reply:
```csharp
        public static Type GetAnalyzerType(string name, string analyzerTypeAsString, string databaseName)
        {
            var analyzerType = LuceneAssembly.GetType(analyzerTypeAsString) ??
                               Type.GetType(analyzerTypeAsString) ??
                               Type.GetType("Raven.Server.Documents.Indexes.Persistence.Lucene.Analyzers." + analyzerTypeAsString) ??
                               LuceneAssembly.GetType("Lucene.Net.Analysis." + analyzerTypeAsString) ??
                               LuceneAssembly.GetType("Lucene.Net.Analysis.Standard." + analyzerTypeAsString) ??
                               AnalyzerCompilationCache.GetAnalyzerType(analyzerTypeAsString, databaseName); // <- added this

            if (analyzerType == null)
                throw new InvalidOperationException($"Cannot find analyzer type '{analyzerTypeAsString}' for field: {name}");

            return analyzerType;
        }
```

Reply:
`MyAnalyzer` comes from where?

Reply:
From the compilation cache? Not sure if I understand you?